src/extension/content-script/batteries/StackOverflow.ts

```diff
@@ -1,41 +1,123 @@
+import axios from "axios";
+import { Battery } from "~/types";
+
 import getOriginData from "../originData";
 import setLightningData from "../setLightningData";
 
 const urlMatcher =
-  /^https:\/\/stackoverflow\.com\/users\/(\d+)\/(\w+)(\?tab=(\w+))?.*/;
+  /^https:\/\/stackoverflow\.com\/(users|questions)\/(\d+)\/(\w+).*/;
+
+const battery = async (): Promise<void> => {
+  const urlParts = document.location.pathname.split("/");
+  const route = urlParts[1];
+  const userOrQuestionId = urlParts[2];
+  let lightningData = null;
 
-const battery = (): void => {
-  const urlParams = new URLSearchParams(window.location.search);
-  const tab = urlParams.get("tab");
+  if (route === "users") {
+    lightningData = await handleProfilePage(userOrQuestionId);
+  }
 
-  if (tab === "profile") {
-    handleProfilePage();
+  if (route === "questions") {
+    lightningData = await handleQuestionPage(userOrQuestionId);
   }
+
+  if (lightningData) setLightningData([lightningData]);
 };
 
-function handleProfilePage() {
-  const aboutElement = document.querySelector<HTMLElement>(
-    ".js-about-me-content"
+function findLightningAddressInText(text: string) {
```

Comment:
let's move this into a helper like `setLightningData`. then this can be reused by other batteries. 

maybe export `setLightningData` and `findLightningAddressInText` from a batteries/helpers.ts file?

Reply:
Done!

Reply:
Done!